crates/models/src/authorizations.rs

```diff
@@ -148,3 +154,29 @@ pub struct UserTaskAuthorization {
     #[serde(default, skip_serializing_if = "String::is_empty")]
     pub shard_id_prefix: String,
 }
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub enum AllowedRole {
```

Comment:
Yeah, I went back and forth on this and ended up going with the more general `/authorize/role` because `/authorize/dekaf` felt like a deviation from the level of generality that everything surrounding it exists in. That is, the rest of the agent API is fairly general and applicable to the system as a whole, whereas `/authorize/dekaf` would be highly specific to Dekaf and nothing else. 

Of course, going down the less general/more specific path has the very nice benefit of not having to worry about how to fit the things you need the API to return into the abstraction, but aren't you worried about this moving the agent API towards becoming a sort of "kitchen sink" where we add new endpoints whenever we need a new one, as opposed to presenting a more coherent interface?

I'm certainly happy to take the `/authorize/dekaf` path here, as like you say it'll enable returning a bunch more related data in one go, and also reduce the security-relevant surface area (as we won't be returning any supabase access tokens, short-lived or no).